nft-client/src/main.rs

```diff
@@ -127,7 +126,7 @@ struct AddressWrapper(types::Address);
 /// Parse a string into an address, by first trying to parse the string as a
 /// contract address string, otherwise try parsing as an account address string.
 impl FromStr for AddressWrapper {
-    type Err = &'static str;
+    type Err = concordium_contracts_common::AccountAddressParseError;
```

Comment:
Shouldn't we try to implement `FromStr` for `Address` directly instead of doing this? And the same for ContractAddress.

Similarly for the DIsplay trait. As long as we document the format it's fine.

Reply:
It would be useful to have this. The hard part is deciding the format for contract address. I know we have discussed this previously, but without a conclusion.